### ISSUE TITLE
Rate Limit Not Working

### ISSUE THREAD

**Issue by FedericoHesse** [curiefense/curiefense#754]
**Describe the bug**
On a clean Curiefense configuration, rate limit is not doing anything. We followed the "Getting Started" section from start to finish but we can't get the rate limit to work at all on its default config of 5/60s limit or any other variant. Mostly the same issue as #658.

**To Reproduce**
1. Follow [Getting Started with Curiefense](https://docs.curiefense.io/installation/getting-started-with-curiefense) tutorial up to [Add Rate Limit](https://docs.curiefense.io/installation/getting-started-with-curiefense#add-rate-limiting).
2. Security Policies -> Path Mapping
3. Rate Limit Rules -> To attach an existing rule, click here.
4. Select "Rate Limit Example Rule 5/60 5 requests per minute"
5. Add
6. Save
7. Publish

**Expected behavior**
As per the documentation: _Wait briefly, and you should see that on the 6th request, the response changes from 403 (which means the ACL Profile is blocking the request) to 503 (which means the Rate Limit is blocking it)._ 

As stated by the guide, _hdr-test_ on DENY gets us unending 403s. If we remove it from DENY, we get unending 200s instead.

**Screenshots**
[Rate Limit Default Config](https://i.imgur.com/17JjLeY.png)
Tag _all_ listed on INCLUDE list.

**Additional context**
- Curiefense v1.4 - Istio Helm installation

**Comment by bartavelle** [comment]
In the `compose` directory, there is a `.env` file. If you set the error level to debug, you will be able to see, in the `curieproxy` logs, what exactly is happening. This that you might be able to see are error messages with regards to configuration loading (only on the first request for each configuration change), errors from the redis server, or really, any other error. That would be really useful to understand what is going on!

**Comment by FedericoHesse** [comment]
We also tried replicating this issue #688 on a new clean install to no success. So far we tried rate limiting by counting IPs and headers, and none of them worked.
Redis is not generating any logs besides the following:
```
5 8:C 10 Mar 2022 22:12:06.279 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
4 8:C 10 Mar 2022 22:12:06.279 # Redis version=6.2.6, bits=64, commit=00000000, modified=0, pid=8, just started
3 8:C 10 Mar 2022 22:12:06.279 # Configuration loaded
2 8:M 10 Mar 2022 22:12:06.280 # WARNING: The TCP backlog setting of 1024 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
1 8:M 10 Mar 2022 22:12:06.280 # Server initialized
```

---

> In the `compose` directory, there is a `.env` file. If you set the error level to debug, you will be able to see, in the `curieproxy` logs, what exactly is happening. This that you might be able to see are error messages with regards to configuration loading (only on the first request for each configuration change), errors from the redis server, or really, any other error. That would be really useful to understand what is going on!

Do you have any example of said logs? Right now we've got several logs flying around and, to be frank, I wouldn't know which one is the one.

**Comment by bartavelle** [comment]
After modifying the `.env` file, you can see them using `docker-compose logs  curieproxy`.

**Comment by FedericoHesse** [comment]
Welp, it was an erroneous variable on our side. We had redis set to localhost but since our istio external was in a different namespace than our curiefense, they weren't communicating at all. We set redis' adress as redis.namespace and it worked instantly. Thanks!

Moving on, and sorry if I should open a new issue instead, but we're now testing the content filter and we haven't found a way to identify what rule we are getting blocked by. For example:

![image (2)](https://user-images.githubusercontent.com/37289694/158495914-4bd92431-ce36-4150-8d36-200149d5ebc1.png)

![image (3)](https://user-images.githubusercontent.com/37289694/158495942-67db949d-cf27-4bb6-b740-54c67fe92aee.png)

Is there a way to find out the ID of the content filter rule or do we have to search for it manually on the UI?

**Comment by FedericoHesse** [comment]
That certainly did the trick, thanks Simon!
Closing now to avoid spamming the ticket with irrelevant content to the original topic.
